metadata-ingestion/src/datahub/ingestion/source/delta_lake/delta_lake_utils.py

```diff
@@ -1,53 +1,29 @@
 import pathlib
-from typing import Optional
+from typing import Optional, Dict
 
 from deltalake import DeltaTable, PyDeltaTableError
 
 from datahub.ingestion.source.delta_lake.config import DeltaLakeSourceConfig
 
 
 def read_delta_table(
-    path: str, delta_lake_config: DeltaLakeSourceConfig
+    path: str, opts: Dict[str, str], delta_lake_config: DeltaLakeSourceConfig
 ) -> Optional[DeltaTable]:
-    delta_table = None
-    try:
-        opts = {}
-        if delta_lake_config.is_s3:
-            if (
-                delta_lake_config.s3 is not None
-                and delta_lake_config.s3.aws_config is not None
-            ):
-                creds = delta_lake_config.s3.aws_config.get_credentials()
```

Comment:
Move calculating `opts` outside so that this doesn't get repeatedly called

Reply:
nice